synapse/handlers/send_email.py

```diff
@@ -109,23 +112,13 @@ def build_sender_factory(**kwargs: Any) -> ESMTPSenderFactory:
         # set to enable TLS.
         factory = build_sender_factory(hostname=smtphost if enable_tls else None)
 
+    endpoint = HostnameEndpoint(
+        reactor, smtphost, smtpport, timeout=30, bindAddress=None
+    )
     if force_tls:
-        reactor.connectSSL(
-            smtphost,
-            smtpport,
-            factory,
-            optionsForClientTLS(smtphost),
-            timeout=30,
-            bindAddress=None,
-        )
-    else:
-        reactor.connectTCP(
-            smtphost,
-            smtpport,
-            factory,
-            timeout=30,
-            bindAddress=None,
-        )
+        factory = TLSMemoryBIOFactory(optionsForClientTLS(smtphost), True, factory)
+
+    await make_deferred_yieldable(endpoint.connect(factory))
```

Comment:
Last comment, sorry. 😄 

I think it would make sense to move the `endpoint = HostnameEndpoint(...)` below the `if force_tls` bit, as then all the code to figure out which "factory" we want to use is next to each other.

Reply:
Heh, that's a great point!